modules/affile/file-opener.h

```diff
@@ -64,7 +65,8 @@ struct _FileOpener
   LogTransport *(*construct_transport)(FileOpener *self, gint fd);
   LogProtoServer *(*construct_src_proto)(FileOpener *self, LogTransport *transport,
                                          LogProtoFileReaderOptions *proto_options);
-  LogProtoClient *(*construct_dst_proto)(FileOpener *self, LogTransport *transport, LogProtoClientOptions *proto_options);
+  LogProtoClient *(*construct_dst_proto)(FileOpener *self, LogTransport *transport, LogProtoClientOptions *proto_options,
+                                         SignalSlotConnector *connector);
```

Comment:
👍🏻 

The `construct_dst_proto` function pointer is for imitating C++ virtual methods in C.
Since you changed the signature of the virtual method, you need to adjust the parameter list in subclasses too.

The compiler warnings show you the missing pieces, you will eventually reach `log_proto_file_writer_new()`.

`modules/affile/named-pipe.c` - we don't want to do anything with the connector here, just fix the signature
`modules/affile/regular-files.c` - this is where it gets interesting

Reply:
Fixed :+1: